app/knights/battle.py

```diff
@@ -0,0 +1,37 @@
+from app.knights.knights import lancelot, mordred, arthur, red_knight
+from app.config.config import KNIGHTS
+
+
+# BATTLE:
+class Battle:
+    @staticmethod
+    def battle(knights_config: dict) -> dict:
+        # 1 Lancelot vs Mordred:
+        lancelot.hp -= mordred.power - lancelot.protection
+        mordred.hp -= lancelot.power - mordred.protection
```

Comment:
Similar to the previous line, `mordred.protection` is accessed directly, which is not defined in the `Knight` class. Ensure that protection is calculated from the armour list.